doc/internal/man3/OSSL_TIME.pod

```diff
@@ -0,0 +1,90 @@
+=pod
+
+=head1 NAME
+
+OSSL_TIME, OSSL_TIME_SECOND, OSSL_TIME_NEVER,
+ossl_time_now, ossl_time_time_to_timeval, ossl_time_compare,
+ossl_time_add, ossl_time_subtract
+- times and durations
+
+=head1 SYNOPSIS
+
+ #include "internal/time.h"
+
+ typedef OSSL_TIME;
+
+ #define OSSL_TIME_SECOND
+ #define OSSL_TIME_NEVER
+
+ OSSL_TIME ossl_time_now(void);
+ void ossl_time_time_to_timeval(OSSL_TIME t, struct timeval *out);
```

Comment:
I'm not sure about `struct timespec`, it would need conditioning.  Another one that might be useful is `time_t`.  I didn't think about the reverse versions which are a good idea.  The big caveat is that these are internal right now and we don't need the other forms.  The only things libssl wants is the current time as a `struct timeval`, so that's what I did.

If we ever made them public, we'd want to revisit the conversions made available.